src/ros_mac_authentication.cpp

```diff
@@ -90,7 +90,7 @@ bool authenticate(rosauth::Authentication::Request &req, rosauth::Authentication
     SHA512((unsigned char *)local_hash.c_str(), local_hash.length(), sha512_hash);
 
     // convert to a hex string to compare
-    char hex[SHA512_DIGEST_LENGTH * 2];
+    char hex[SHA512_DIGEST_LENGTH * 2 + 1];
```

Comment:
I don't see why this is required, chars are added on `i * 2` and `i * 2 + 1`. So `SHA512_DIGEST_LENGTH-1=63 * 2 + 1 = 127` is the last index. Seems in bound of a size `128` array. What is your reasoning behind this?

Reply:
If it is due to the trailing null character, shouldn't we add `'\0'` to the last index in the array?

Reply:
`sprintf(pos, "%02x", 0x12);` writes bytes like:
```
pos[0] = '1';
pos[1] = '2';
pos[2] = '\0';
```
So, in this case length of `pos` must be `1 * 2 + 1` instead of `1 * 2`.

On gcc, this node built with `-fstack-protector-strong` (which is often used for security oriented system to protect from buffer overrun related vulnerability) detects buffer overrun, and abort.

Reply:
Ok, but should't you add the null character then?

Reply:
Clear, thanks